### PR TITLE
New asm syntax and other misc bits.

### PR DESCRIPTION
**Pull request by vext01**
You will probably want to look at individual commits.

A rustfmt PR will follow.

### CODE REVIEW

**Comment by vext01** [comment]
bors ping

**Comment by vext01** [comment]
Please don't merge yet, I need to add this repo to the buildbot.

**Comment by vext01** [comment]
This is ready.

**Comment by ptersilie** [comment]
Looks good to me. Anything to squash? If so please do.

**Comment by vext01** [comment]
I'd like to leave this all separate, so no squashing.

**Comment by ptersilie** [comment]
Alright then.

bors r+

**Comment by vext01** [comment]
Had to tweak some repo settings.

Please try again.

File: build.rs [hunk 1]
```diff
@@ -71,81 +99,86 @@ fn cpu_supports_pt() -> bool {
...
-            : "i" (LEAF), "i" (SUBPAGE)
-            : "eax", "ecx", "edx"
-            : "volatile");
+        asm!(
+              "cpuid",
+              inout("eax") LEAF => _,
+              inout("ecx") SUBPAGE => _,
+              lateout("ebx") ebx_out,
+              lateout("edx") _,
+        );
```
Comment (bjorn3) [inline build.rs:108]: You could use [`core::arch::x86_64::__cpuid_count`](https://doc.rust-lang.org/stable/core/arch/x86_64/fn.__cpuid_count.html) instead.

Comment (vext01) [reply build.rs:108]: Nice! How did I not know about this?

Will do this in a separate PR, I think.

Thanks

**Pull request merged**